### PR TITLE
bugfix/stock-default-axis-options

### PR DESCRIPTION
**Pull request by khlieng**
Fixed dynamically added axis defaults in stock, axes dynamically added to stock charts got different defaults than the ones added through the chart constructor.

### CODE REVIEW

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks for the PR!

We need some tests too. Does the `StockChart.createAxis` function actually run at all? As far as I can see, the `StockChart` class isn't used. The `stockChart` factory returns an instance of `Chart`. And `Highcharts.StockChart` is a reference to the `stockChart` factory.

@bre1470 FYI

**Comment by khlieng** [comment]
So I guess most of the factory function should be moved into an init function now that theres a real class, and then change `Highcharts.StockChart` to the class?

File: ts/Core/Chart/StockChart.ts [hunk 1]
```diff
@@ -1158,8 +1169,8 @@ function getDefaultAxisOptions(type: string, options: Highcharts.AxisOptions): H
  *
  * */
 
-(H as any).stockChart = stockChart;
-(H as any).StockChart = StockChart;
+H.stockChart = stockChart;
+H.StockChart = StockChart;
```
Comment (bre1470) [inline ts/Core/Chart/StockChart.ts:1173]: This should be moved to the master file for stock (modules/stock). All files except masters have to work without Highcharts namespace and automatic modifications to become compatible with ES6 tree shaking (async). And it is okay to have any casts in the master file. No need to define the namespace there.

File: ts/masters/modules/stock.src.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@
  * License: www.highcharts.com/license
  */
 'use strict';
+import Highcharts from '../highcharts.src.js';
```
Comment (bre1470) [inline ts/masters/modules/stock.src.ts:13]: That needs to be taken from `'../../Core/Globals.js'` otherwise it bloats the product size.

**Review by TorsteinHonsi — APPROVED** [review]

**Comment by pawelfus** [comment]
`xAxis.type` still fails: https://jsfiddle.net/BlackLabel/g7cfja6q/
Related ticket: https://github.com/highcharts/highcharts/issues/8603 

@khlieng

**Pull request merged**